apollo-federation/src/supergraph/mod.rs

```diff
@@ -709,6 +713,21 @@ fn extract_object_type_content(
             message: "@join__implements should exist for a fed2 supergraph".to_owned(),
         })?;
 
+    let context = supergraph_schema
+        .metadata()
+        .and_then(|metadata| metadata.for_identity(&Identity::context_identity()))
+        .and_then(|context_link| CONTEXT_VERSIONS.find(&context_link.url.version))
+        .and_then(|context_spec_def| {
+            if let Some(name_in_schema) = context_spec_def
+                .context_directive_name_in_schema(supergraph_schema)
+                .ok()
+            {
+                return Some((context_spec_def, name_in_schema));
+            }
+            None
```

Comment:
Two thoughts here. If `context_directive_name_in_schema` fails, we are silencing that error. What could cause that to error? Second, IMO, mapping the output from `context_directive_name_in_schema` would be more readable:
```suggestion
            context_spec_def
                .context_directive_name_in_schema(supergraph_schema)
                .ok()
                .map(|name_in_schema| (context_spec_def, name_in_schema))
```
If we want to bubble up the error, just remove the `.ok()` then call `.transpose()?` after the last `.and_then` to invert the `Option<Result>` to a `Result<Option>` then bubbling up the error with the `?` operator.